cli/options.go

```diff
@@ -1,190 +1,515 @@
-	fs.BoolVar(&defaults.NATSPubSub.DontRandomizeServers, "nats_dont_randomize_servers", defaults.NATSPubSub.DontRandomizeServers, "")
-
-	fs.StringVar(&defaults.RPC.Host, "rpc_host", "localhost:50051", "")
-	fs.IntVar(&defaults.RPC.Concurrency, "rpc_concurrency", 28, "")
-	fs.BoolVar(&defaults.RPC.EnableTLS, "rpc_enable_tls", false, "")
-	fs.IntVar(&defaults.RPC.MaxRecvSize, "rpc_max_call_recv_size", 0, "")
-	fs.IntVar(&defaults.RPC.MaxSendSize, "rpc_max_call_send_size", 0, "")
-	fs.StringVar(&headers, "headers", "cookie", "")
-
-	fs.IntVar(&defaults.WS.ReadBufferSize, "read_buffer_size", 1024, "")
-	fs.IntVar(&defaults.WS.WriteBufferSize, "write_buffer_size", 1024, "")
-	fs.Int64Var(&defaults.WS.MaxMessageSize, "max_message_size", 65536, "")
-	fs.BoolVar(&defaults.WS.EnableCompression, "enable_ws_compression", false, "")
-	fs.StringVar(&defaults.WS.AllowedOrigins, "allowed_origins", "", "")
-
-	fs.IntVar(&defaults.DisconnectQueue.Rate, "disconnect_rate", 100, "")
-	fs.IntVar(&defaults.DisconnectQueue.ShutdownTimeout, "disconnect_timeout", 5, "")
-	fs.BoolVar(&defaults.DisconnectorDisabled, "disable_disconnect", false, "")
-
-	fs.StringVar(&defaults.LogLevel, "log_level", "info", "")
-	fs.StringVar(&defaults.LogFormat, "log_format", "text", "")
-	fs.BoolVar(&debugMode, "debug", false, "")
-
-	fs.BoolVar(&defaults.Metrics.Log, "metrics_log", false, "")
-	fs.IntVar(&defaults.Metrics.RotateInterval, "metrics_rotate_interval", 0, "")
-	fs.IntVar(&defaults.Metrics.LogInterval, "metrics_log_interval", -1, "")
-	fs.StringVar(&defaults.Metrics.LogFormatter, "metrics_log_formatter", "", "")
-	fs.StringVar(&defaults.Metrics.HTTP, "metrics_http", "", "")
-	fs.StringVar(&defaults.Metrics.Host, "metrics_host", "", "")
-	fs.IntVar(&defaults.Metrics.Port, "metrics_port", 0, "")
-
-	fs.IntVar(&defaults.App.PingInterval, "ping_interval", 3, "")
-	fs.StringVar(&defaults.App.PingTimestampPrecision, "ping_timestamp_precision", "s", "")
-	fs.IntVar(&defaults.App.StatsRefreshInterval, "stats_refresh_interval", 5, "")
-	fs.IntVar(&defaults.App.HubGopoolSize, "hub_gopool_size", 16, "")
-
-	fs.StringVar(&defaults.JWT.Secret, "jwt_id_key", "", "")
-	fs.StringVar(&defaults.JWT.Param, "jwt_id_param", "jid", "")
-	fs.BoolVar(&defaults.JWT.Force, "jwt_id_enforce", false, "")
-
-	fs.StringVar(&defaults.Rails.TurboRailsKey, "turbo_rails_key", "", "")
-	fs.StringVar(&defaults.Rails.CableReadyKey, "cable_ready_key", "", "")
-
-	// CLI vars
-	fs.BoolVar(&showHelp, "h", false, "")
-	fs.BoolVar(&showVersion, "v", false, "")
-}
-
-// Config returns CLI configuration
-func Config(args []string) (config.Config, error) {
-	if err := fs.Parse(args); err != nil {
-		return config.Config{}, err
-	}
+func redisCLIFlags(c *config.Config) []cli.Flag {
+	return withDefaults(redisCategoryDescription, []cli.Flag{
+		&cli.StringFlag{
+			Name:        "redis_url",
+			Usage:       "Redis url",
+			Value:       c.Redis.URL,
+			EnvVars:     []string{"ANYCABLE_REDIS_URL", "REDIS_URL"},
```

Comment:
What about prepending prefixed env to the list of passed EnvVars if they defined?

This way, all the prefixed envs would be handled automatically in one place.